File: Source/Templates/GtkSharp.Template.FSharp/content/GtkSharp.Dialog.FSharp/Gtk_Dialog.fs
namespace GtkNamespace

open Gtk

type Gtk_Dialog (builder : Builder) as this =
    inherit Dialog(builder.GetRawOwnedObject("Gtk_Dialog"))
    do
        this.DefaultResponse <- ResponseType.Cancel;
        this.Response.Add(fun _ ->
            this.Hide();
        )

    new() = new Gtk_Dialog(new Builder("Gtk_Dialog.glade"))

File: Source/Workload/GtkSharp.Workload.Template.CSharp/content/GtkSharp.Dialog.CSharp/Gtk_Dialog.cs
using System;
using Gtk;
using UI = Gtk.Builder.ObjectAttribute;

namespace GtkNamespace
{
    class Gtk_Dialog : Dialog
    {
        public Gtk_Dialog() : this(new Builder("Gtk_Dialog.glade")) { }

        private Gtk_Dialog(Builder builder) : base(builder.GetRawOwnedObject("Gtk_Dialog"))
        {
            builder.Autoconnect(this);
            DefaultResponse = ResponseType.Cancel;

            Response += Dialog_Response;
        }

        private void Dialog_Response(object o, ResponseArgs args)
        {
            Hide();
        }
    }
}
